code/modules/robotics/robot/robot_docking_station.dm

```diff
@@ -818,15 +870,60 @@ TYPEINFO(/obj/machinery/recharge_station)
 			R.update_appearance()
 			. = TRUE
 
+		if("repair-fuel-drone")
+			if (!isdrone(occupant))
+				return
+			var/mob/living/silicon/drone/D = src.occupant
+			if (src.reagents.get_reagent_amount("fuel") < 1)
+				boutput(user, "<span class='alert'>Not enough welding fuel for repairs.</span>")
+				return
+			if ((!issilicon(user) && (BOUNDS_DIST(user, src) > 0)) || user.stat)
+				return
+			var/usage = min(src.reagents.get_reagent_amount("fuel"), D.bruteloss)
+			if (usage < 1)
+				return
+			D.HealDamage("all", usage, 0)
+			src.reagents.remove_reagent("fuel", usage)
+			D.update_appearance()
+			. = TRUE
+
+		if("repair-wiring-drone")
+			if (!isdrone(occupant))
+				return
+			var/mob/living/silicon/drone/D = src.occupant
+			if (src.cabling < 1)
+				boutput(user, "<span class='alert'>Not enough wiring for repairs.</span>")
+				return
+			if ((!issilicon(user) && (BOUNDS_DIST(user, src) > 0)) || user.stat)
+				return
+			var/usage =  min(src.cabling, D.fireloss)
+			if (usage < 1)
+				return
+			D.HealDamage("all", 0, usage)
+			src.cabling -= usage
+			if (src.cabling < 0)
+				src.cabling = 0
+			D.update_appearance()
+			. = TRUE
+
 		if("module-install")
 			if (!isrobot(src.occupant))
 				return
 			var/mob/living/silicon/robot/R = src.occupant
 			var/moduleRef = params["moduleRef"]
 			if(moduleRef)
 				var/obj/item/robot_module/module = locate(moduleRef) in src.modules
+				if (isdrone(src.occupant) && module.moduletype != "drone")
```

Comment:
Hmm.. It would probably make things cleaner if it did, yeah. A lot of things are shared between the two types which would be a lot easier to work with to if they were stored like that? I think.
I'm not the most experienced programmer, so take what I'm saying with a grain of salt, hwhaha.